[depends] freetype 2.7.1 requires specific harfbuzz version

To reproduce:
```
cd depends
make
cd ..
./configure --prefix=`pwd`/depends/x86_64-pc-linux-gnu
```

Problem:
```
checking for static Qt plugins: -lqminimal... no
configure: WARNING: Could not resolve: -lqminimal; bitcoin-qt frontend will not be built
checking for static Qt plugins: -lqxcb -lxcb-static... no
checking whether to build Bitcoin Core GUI... no (Qt5)
```

The actual problem is a linking issue when running the configuration check. From `config.log`:
```
/usr/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../../lib/libharfbuzz.so: undefined reference to `FT_Done_MM_Var'
```
[(Full snippet of relevant section in config.log)](https://github.com/bitcoin/bitcoin/files/2437677/config-log-snippet.txt)

I run Arch Linux, which currently uses harfbuzz 1.9.0. Downgrading harfbuzz to to 1.7.4 makes `configure` happy again (1.7.5 is the first version that fails) but that defeats the purpose of the `depends` build. Our `depends` system includes freetype 2.7.1 which however requires a specific version (range) of harfbuzz.

It seems that one naive way to resolve that issue is to include harfbuzz in `depends` but actually I don't know about all the magic and trade-offs going on the build system.

This may be related to #13001.

--- a/depends/packages/freetype.mk
+++ b/depends/packages/freetype.mk
@@ -5,7 +5,7 @@ $(package)_file_name=$(package)-$($(package)_version).tar.bz2
 $(package)_sha256_hash=3a3bb2c4e15ffb433f2032f50a5b5a92558206822e22bfe8cbe339af4aa82f88
 
 define $(package)_set_vars
-  $(package)_config_opts=--without-zlib --without-png --disable-static
+  $(package)_config_opts=--without-zlib --without-png --without-harfbuzz --without-bzip2 --disable-static
   $(package)_config_opts_linux=--with-pic
 endef
 

--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -64,6 +64,7 @@ $(package)_config_opts += -prefix $(host_prefix)
 $(package)_config_opts += -qt-libpng
 $(package)_config_opts += -qt-libjpeg
 $(package)_config_opts += -qt-pcre
+$(package)_config_opts += -qt-harfbuzz
 $(package)_config_opts += -system-zlib
 $(package)_config_opts += -static
 $(package)_config_opts += -silent

